reactos/drivers/usb/usbport/roothub.c

```diff
@@ -0,0 +1,1019 @@
+        RH_DeviceDescriptor->bDescriptorType = USB_DEVICE_DESCRIPTOR_TYPE;
+        RH_DeviceDescriptor->bcdUSB = 0x100;
+        RH_DeviceDescriptor->bDeviceClass = USB_DEVICE_CLASS_HUB;
+        RH_DeviceDescriptor->bDeviceSubClass = 0x01;
+        RH_DeviceDescriptor->bDeviceProtocol = 0x00;
+        RH_DeviceDescriptor->bMaxPacketSize0 = 0x08;
+        RH_DeviceDescriptor->idVendor = FdoExtension->VendorID;
+        RH_DeviceDescriptor->idProduct = FdoExtension->DeviceID;
+        RH_DeviceDescriptor->bcdDevice = FdoExtension->RevisionID;
+        RH_DeviceDescriptor->iManufacturer = 0x00;
+        RH_DeviceDescriptor->iProduct = 0x00;
+        RH_DeviceDescriptor->iSerialNumber = 0x00;
+        RH_DeviceDescriptor->bNumConfigurations = 0x01;
+
+        RH_ConfigurationDescriptor = &PdoExtension->RootHubDescriptors->ConfigDescriptor;
+
+        RH_ConfigurationDescriptor->bLength = sizeof(USB_CONFIGURATION_DESCRIPTOR);
+        RH_ConfigurationDescriptor->bDescriptorType = USB_CONFIGURATION_DESCRIPTOR_TYPE;
+
+        RH_ConfigurationDescriptor->wTotalLength = sizeof(USB_CONFIGURATION_DESCRIPTOR) +
+                                                   sizeof(USB_INTERFACE_DESCRIPTOR) +
+                                                   sizeof(USB_ENDPOINT_DESCRIPTOR);
+
+        RH_ConfigurationDescriptor->bNumInterfaces = 0x01;
+        RH_ConfigurationDescriptor->bConfigurationValue = 0x01;
+        RH_ConfigurationDescriptor->iConfiguration = 0x00;
+        RH_ConfigurationDescriptor->bmAttributes = 0x40;
+        RH_ConfigurationDescriptor->MaxPower = 0x00;
+
+        RH_InterfaceDescriptor = &PdoExtension->RootHubDescriptors->InterfaceDescriptor;
+
+        RH_InterfaceDescriptor->bLength = sizeof(USB_INTERFACE_DESCRIPTOR);
+        RH_InterfaceDescriptor->bDescriptorType = USB_INTERFACE_DESCRIPTOR_TYPE;
+        RH_InterfaceDescriptor->bInterfaceNumber = 0x00;
+        RH_InterfaceDescriptor->bAlternateSetting = 0x00;
+        RH_InterfaceDescriptor->bNumEndpoints = 0x01;
+        RH_InterfaceDescriptor->bInterfaceClass = USB_DEVICE_CLASS_HUB;
+        RH_InterfaceDescriptor->bInterfaceSubClass = 0x01;
+        RH_InterfaceDescriptor->bInterfaceProtocol = 0x00;
+        RH_InterfaceDescriptor->iInterface = 0x00;
+
+        RH_EndPointDescriptor = &PdoExtension->RootHubDescriptors->EndPointDescriptor;
+
+        RH_EndPointDescriptor->bLength = sizeof(USB_ENDPOINT_DESCRIPTOR);
+        RH_EndPointDescriptor->bDescriptorType = USB_ENDPOINT_DESCRIPTOR_TYPE;
+        RH_EndPointDescriptor->bEndpointAddress = 0x81;
+        RH_EndPointDescriptor->bmAttributes = USB_ENDPOINT_TYPE_INTERRUPT; // SCE endpoint
+        RH_EndPointDescriptor->wMaxPacketSize = 0x0008;
+        RH_EndPointDescriptor->bInterval = 0x0C; // 12 msec
+
+        RH_HubDescriptor = &PdoExtension->RootHubDescriptors->Descriptor;
+
+        RH_HubDescriptor->bDescriptorLength = sizeof(USB_HUB_DESCRIPTOR) + 2 * NumMaskByte;
+        RH_HubDescriptor->bDescriptorType = 0x29;
```

Comment:
Can you add a comment mentioning USB_20_HUB_DESCRIPTOR_TYPE? (this is yet another define missing from our headers)